### PR TITLE
Fix docstrings for pandas.Period.month

### PR DESCRIPTION
**Pull request by uditbaliyan**
Part of https://github.com/pandas-dev/pandas/issues/59458

Addresses:
pandas.Period.month having SA01
pandas.Period.asfreq SA01

Changes:
Added See Also section .
Added Returns section .
Added extended section .
Added notes section .
Added examples section .
Removed  "pandas.Period.month SA01" and "pandas.Period.asfreq SA01" from code_checks.sh

### CODE REVIEW

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -2011,15 +2011,61 @@ cdef class _Period(PeriodMixin):
 
     @property
     def month(self) -> int:
-        """
-        Return the month this Period falls on.
+    """
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/period.pyx:2014]: in indent was correct before

Comment (uditbaliyan) [reply pandas/_libs/tslibs/period.pyx]: Thanks for the feedback. I've updated the indentation of the function in `pandas/_libs/tslibs/period.pyx`. 
 If you have any further suggestions or if there's anything specific you’d like me to address, please let me know!

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -2014,12 +2014,58 @@ cdef class _Period(PeriodMixin):
...
+
+        >>> period = pd.Period('nan', 'M')
+        >>> period.month
+        nan
+
+        Handle a case where the Period object is invalid or empty, which results in `NaN`:
+
+        >>> period = pd.Period('nan', 'M')
+        >>> period.month
+        nan
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/period.pyx:2062]: repeated?

**Comment by MarcoGorelli** [comment]
thnx, please check ci failure

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @uditbaliyan

**Pull request merged**